homeassistant/components/homematicip_cloud/config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Config flow to configure HomematicIP Cloud."""
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN, _LOGGER, CONF_ACCESSPOINT, CONF_PIN, CONF_NAME
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class HomematicipCloudFlowHandler(data_entry_flow.FlowHandler):
+    """Config flow HomematicIP Cloud."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize HomematicIP Cloud config flow."""
+        self.hmip_auth = None
+        self.hmip_apid = None
+        self.hmip_name = None
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        from homematicip.aio.auth import AsyncAuth
+        from homematicip.base.base_connection import HmipConnectionError
+        errors = {}
+
+        if user_input is not None:
+            if user_input[CONF_ACCESSPOINT]:
+                self.hmip_apid = user_input[CONF_ACCESSPOINT]
+                self.hmip_name = user_input[CONF_NAME]
+                _LOGGER.info("Create new authtoken for %s", self.hmip_apid)
+
+                # Create new authtoken for the accesspoint
+                websession = aiohttp_client.async_get_clientsession(self.hass)
+                self.hmip_auth = AsyncAuth(self.hass.loop, websession)
+                try:
+                    await self.hmip_auth.init(self.hmip_apid)
+                except HmipConnectionError:
+                    return self.async_abort(reason='conection_aborted')
+                if user_input[CONF_PIN]:
+                    self.hmip_auth.pin = user_input[CONF_PIN]
+                try:
+                    state = await self.hmip_auth.connectionRequest(
+                        'HomeAssistant')
+                except HmipConnectionError:
+                    if state['errorCode'] == 'INVALID_PIN':
```

Comment:
This code cannot work. If `hmip_auth.connectionRequest` blows up, `state` is not assigned.